org.eclipse.scanning.points/src/org/eclipse/scanning/jython/JythonInterpreterManager.java

```diff
@@ -0,0 +1,243 @@
+			loader = createJythonClassLoader(state.getClassLoader());   // Don't clobber their working.		
+    	}
+	   	state.setClassLoader(loader);
+    	
+     	setJythonPaths(state);       // Tries to ensure that enough of jython is on the path
+     	setSpgGeneratorPaths(state, bundleNames); // Adds the scripts directory from points
+	   	Py.setSystemState(state);
+ 
+	   	configuredState = state;
+	}
+
+	private static void setSpgGeneratorPaths(PySystemState state, String... bundleNames) {
+		
+        File location = getBundleLocation("org.eclipse.scanning.points");
+        state.path.add(new PyString(location.getAbsolutePath() + "/scripts/"));
+        
+        location = getBundleLocation("org.eclipse.scanning.sequencer");
+        state.path.add(new PyString(location.getAbsolutePath() + "/scripts/"));
+
+        if (bundleNames!=null) {
+        	for (String bundle : bundleNames) {
+                location = getBundleLocation(bundle);
+                state.path.add(new PyString(location.getAbsolutePath() + "/scripts/"));
+			}
+        }
+	}
+
+
+	private static void initializePythonPath(ClassLoader loader) {
+		try {
+	    	String jythonBundleName = System.getProperty("org.eclipse.scanning.jython.osgi.bundle.name", "uk.ac.diamond.jython");
+	        File loc = getBundleLocation(jythonBundleName); // TODO Name the jython OSGi bundle without Diamond in it!
+	        if (loc ==null) return;
+	        File jythonDir = find(loc, "jython");
+	        if (jythonDir ==null) return;
+	        
+	        Properties props = new Properties();
+	    	props.setProperty("python.home", jythonDir.getAbsolutePath());
+	    	props.setProperty("python.console.encoding", "UTF-8"); // Used to prevent: console: Failed to install '': java.nio.charset.UnsupportedCharsetException: cp0.
+	    	props.setProperty("python.options.showJavaExceptions", "true");
+	    	props.setProperty("python.verbose", "warning");
+
+	    	Properties preprops = System.getProperties();
+	    	
+	    	PySystemState.initialize(preprops, props, new String[]{}, loader);
+	    
+	    } catch (Throwable ne) {
+	    	System.out.print("Problem loading jython bundles!");
+	    	ne.printStackTrace();
+	    	logger.debug("Problem loading jython bundles!", ne);
+	    }
+	}
+
+
+	private static ClassLoader createJythonClassLoader(ClassLoader classLoader) {
+		
+    	ClassLoader jythonClassloader = ScanPointGeneratorFactory.class.getClassLoader();
+    	
+    	try { // For non-unit tests, attempt to use the OSGi classloader of this bundle.
+    		String jythonBundleName = System.getProperty("org.eclipse.scanning.jython.osgi.bundle.name", "uk.ac.diamond.jython");
+    		CompositeClassLoader composite = new CompositeClassLoader(classLoader);
+   	  	    // Classloader for org.eclipse.scanning.points
+    		composite.addLast(ScanPointGeneratorFactory.class.getClassLoader());
+    		addLast(composite, jythonBundleName);
+    		jythonClassloader = composite;
+     		
+    	} catch (Throwable ne) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1181)